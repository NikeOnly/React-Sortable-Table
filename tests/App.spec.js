import React from 'react'
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme'
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import ConnectedApp, {App} from '../src/App';

describe('App', () => {
    test('match snapshot', () => {
        const tree = renderer.create(<App />).toJSON()
        expect(tree).toMatchSnapshot();
    })

    test('render connected component', () => {
        const initialState = {tableField: {idToField: new Map}}
        const mockStore = configureStore()
        const store = mockStore(initialState)
        const container = shallow(<ConnectedApp store={store} /> )

        expect(toJson(container)).toMatchSnapshot();
        expect(container.prop('idToField')).toEqual(initialState.idToField)
    })
})
