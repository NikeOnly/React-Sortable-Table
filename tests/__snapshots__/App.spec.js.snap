// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App match snapshot 1`] = `
<div>
  <h3>
    Simple form
  </h3>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group "
    >
      <label
        htmlFor="firstName"
      >
        First Name
      </label>
      <input
        className="form-control"
        id="firstName"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="First Name"
        type="text"
        value=""
      />
      <span
        className="help-block"
      />
    </div>
    <div
      className="form-group "
    >
      <label
        htmlFor="lastName"
      >
        Last Name
      </label>
      <input
        className="form-control"
        id="lastName"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Last Name"
        type="text"
        value=""
      />
      <span
        className="help-block"
      />
    </div>
    <div
      className="form-group "
    >
      <label
        htmlFor="phone"
      >
        Phone
      </label>
      <input
        className="form-control"
        id="phone"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Phone"
        type="tel"
        value=""
      />
      <span
        className="help-block"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="radio-inline"
      >
        <input
          checked={true}
          onChange={[Function]}
          type="radio"
          value="Male"
        />
        Male
      </label>
      <label
        className="radio-inline"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="radio"
          value="Female"
        />
        Female
      </label>
    </div>
    <div
      className="form-group "
    >
      <label
        htmlFor="age"
      >
        Age
      </label>
      <input
        className="form-control"
        id="age"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Age"
        type="number"
        value=""
      />
      <span
        className="help-block"
      />
    </div>
    <button
      className="btn btn-default"
      type="submit"
    >
      Submit
    </button>
    <div
      className="has-error"
    >
      <span
        className="help-block"
      >
        
      </span>
    </div>
  </form>
  <table
    className="table table-bordered main-table"
  >
    <thead>
      <tr>
        <th
          onClick={[Function]}
        >
          <span
            className="data-table-header-span"
          >
            #
          </span>
          <i
            className="sort-icon icon-hidden material-icons"
          >
            keyboard_arrow_down
          </i>
        </th>
        <th
          onClick={[Function]}
        >
          <span
            className="data-table-header-span"
          >
            First Name
          </span>
          <i
            className="sort-icon icon-hidden material-icons"
          >
            keyboard_arrow_down
          </i>
        </th>
        <th
          onClick={[Function]}
        >
          <span
            className="data-table-header-span"
          >
            Last Name
          </span>
          <i
            className="sort-icon icon-hidden material-icons"
          >
            keyboard_arrow_down
          </i>
        </th>
        <th
          onClick={[Function]}
        >
          <span
            className="data-table-header-span"
          >
            Phone
          </span>
          <i
            className="sort-icon icon-hidden material-icons"
          >
            keyboard_arrow_down
          </i>
        </th>
        <th
          onClick={[Function]}
        >
          <span
            className="data-table-header-span"
          >
            Gender
          </span>
          <i
            className="sort-icon icon-hidden material-icons"
          >
            keyboard_arrow_down
          </i>
        </th>
        <th
          onClick={[Function]}
        >
          <span
            className="data-table-header-span"
          >
            Age
          </span>
          <i
            className="sort-icon icon-hidden material-icons"
          >
            keyboard_arrow_down
          </i>
        </th>
        <th />
      </tr>
    </thead>
  </table>
</div>
`;

exports[`App render connected component 1`] = `
<App
  addField={[Function]}
  deleteField={[Function]}
  fields={Map {}}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;
